include/seastar/net/api.hh

```diff
@@ -444,6 +444,28 @@ public:
     bool supports_ipv6() const;
 };
 
+class statistics {
+    uint64_t _bytes_sent;
+    uint64_t _bytes_received;
+public:
+    statistics(uint64_t bytes_sent, uint64_t bytes_received)
+        : _bytes_sent(bytes_sent), _bytes_received(bytes_received) {}
+    uint64_t bytes_sent() const {
+        return _bytes_sent;
+    };
+    uint64_t bytes_received() const {
+        return _bytes_received;
+    };
+};
```

Comment:
Minor, but such pure data classes are better as structs with exposed data members and inline initializers for all data members. Less noisy.

Reply:
We did investigate the struct option on our end, but it added 1 instruction to the hot path and we knew Avi-instruction-counter would be reviewing this we went with the separateed approach.

The extra instruction is probably only on x86 because at 16 bytes the struct can't be indexed with indexed addressing which maxes out at * 8, so it needs a separate `shl` do to the indexing.

Reply:
How can struct/class possibly matter here? The layout is the same.

Reply:
Sorry @avikivity I didn't look carefully enough, I though you were referring to the separate read/write TLS variables, which are not bundled into a struct, not this one which is used to expose the metrics.

Reply:
changed